core/server/data/meta/url.js

```diff
@@ -1,24 +1,31 @@
 var schema = require('../schema').checks,
     config = require('../../config');
 
+function sanitizeAmpUrl(url) {
```

Comment:
i've tested this code in the browser. `config.urlFor` did not return the amp postfix in case of a post.

I've seen your tests, which simulate having amp postfix inside the url property, but i am not sure they simulate a real scenario. `post.url` is created without knowing the current url.